backend/src/routes/api/notebooks/notebookUtils.ts

```diff
@@ -139,31 +161,28 @@ export const createNotebook = async (
     throw error;
   });
 
-  await new Promise((r) => setTimeout(r, 500));
-  const route = await fastify.kube.customObjectsApi
-    .getNamespacedCustomObject('route.openshift.io', 'v1', namespace, 'routes', notebookName)
-    .then((response) => response.body as Route)
-    .catch((res) => {
-      const e = res.response.body;
-      const error = createCustomError('Error fetching Notebook Route', e.message, e.code);
-      fastify.log.error(error);
-      throw error;
-    });
-
-  const patch: RecursivePartial<Notebook> = {
-    metadata: {
-      annotations: {
-        'opendatahub.io/link': `https://${route.spec.host}/notebook/${namespace}/${notebookName}`,
-      },
-    },
-  };
-
-  return await patchNotebook(fastify, patch, namespace, notebookName).catch((res) => {
-    const e = res.response.body;
-    const error = createCustomError('Error adding route data to Notebook', e.message, e.code);
-    fastify.log.error(error);
-    throw error;
-  });
+  let route: Route;
+  let fetchTime = 10;
+  while (fetchTime > 0) {
+    try {
+      route = await getRoute(fastify, namespace, notebookName);
+      return await patchNotebookRoute(fastify, route, namespace, notebookName);
+    } catch (e) {
+      // if the route is not created yet
+      // wait for 1 sec and fetch time minus 1
+      if (e.response?.statusCode === 404) {
```

Comment:
I think you want `e.code` once you convert the Route over -- this would not handle if `patchNotebookRoute` got a 404... which is probably good -- since you're patching a very expected to-be-there object (the Notebook).

Perhaps you want to move this logic into just a `.catch` of the `getRoute`?